manager/controlapi/secret.go

```diff
@@ -245,9 +242,11 @@ func validateSecretSpec(spec *api.SecretSpec) error {
 	if err := validateConfigOrSecretAnnotations(spec.Annotations); err != nil {
 		return err
 	}
-
-	if len(spec.Data) >= MaxSecretSize || len(spec.Data) < 1 {
-		return grpc.Errorf(codes.InvalidArgument, "secret data must be larger than 0 and less than %d bytes", MaxSecretSize)
+	if spec.Driver.Name != "" {
```

Comment:
Thanks @cpuguy83, I made this type nullable, and if defined, validation will throw an error if no name is specified.